src/util.ts

```diff
@@ -26,3 +28,17 @@ export function verboseError(...args: any[]) {
         console.error(args);
     }
 }
+
+export function getUserId(username: string) {
+    if (!username) {
+        throw new Error("Missing argument for username");
+    }
+    const result = execSync(`id -i ${username}`)?.toString();
```

Comment:
Per [docs](https://nodejs.org/api/child_process.html#child_processexecsynccommand-options):

> Never pass unsanitized user input to this function. Any input containing shell metacharacters may be used to trigger arbitrary command execution.

Can you switch to, e.g., [spawnSync](https://nodejs.org/api/child_process.html#child_processspawnsynccommand-args-options) instead which (with shell off - the default) and arguments in a separate array should be a tad more secure?

(Who know what siblings little Bobby Tables](https://xkcd.com/327/) has).

Reply:
good point. That's exactly what I was worried about when I mentioned the sanitising in the meeting :) I guess regex is probably enough sanitation, but `spawnSync` works better I think.